### PR TITLE
fix: Passing architecture file as flag

### PR DESCRIPTION
**Pull request by DanGiles**

### CODE REVIEW

**Comment by DanGiles** [comment]
Still working progress. Need to define a global struct.

**Review by themkots — COMMENTED** [review]
Looks OK

**Review by themkots — COMMENTED** [review]
...

**Comment by themkots** [comment]
I just pushed another commit, `a6eeb69`
```
* a6eeb69 - (HEAD, origin/dg-filename-parameter, dg-filename-parameter) Edited swift_fof.c to match target file updates of swift.c (44 seconds ago) <Themis Kotsialos>
* e4411af - Prevent destruction and empty regeneration of archdat structure. (4 days ago) <Themis Kotsialos>
* 37ee779 - Fixed populating partition struct with scotch target arch file before calling initial partitioning. (6 days ago) <Themis Kotsialos>
* f599c38 - Fixed picking scotch when also have metis/parmetis (7 days ago) <Themis Kotsialos>
* c396756 - Fixed picking scotch when also have metis/parmetis (10 days ago) <Themis Kotsialos>
* 3ff309d - Fixed prevailing SCOTCH invocation with configure script; made ArchData globally visible. (12 days ago) <Themis Kotsialos>
* ca8302e - fix: Passing architecture file as flag (8 weeks ago) <Dan Giles>
```

**Comment by themkots** [comment]
This last commit https://github.com/UCL/swiftsim/pull/27/commits/9c55cd191c28abee0cc7a3936d3155dd91b7f733 takes care of the overflow error that was bothering me -- now scotch graphCheck does not complain about overflow in all cases checked [s32,s64,slocal]

**Review by themkots — APPROVED** [review]
I think that this is OK for this branch.

**Comment by themkots** [comment]
Shall we merge this to master?

**Comment by themkots** [comment]
**!! Remember to sync our master with their master first !!**

**Comment by themkots** [comment]
git diff dg-filename-parameter master --stat | wc -l
121

many files have changed, including `swift.c`, `swift_fof.c` and many in `/src`

File: src/partition.c [hunk 1]
```diff
@@ -1555,7 +1553,7 @@ static void pick_scotch(int nodeID, struct space *s, int nregions,
           failed++;
         }
         if (weights_e[k] < 1) {
-          message("Used edge weight out of range: %" PRIDX, weights_e[k]);
+          message("Used edge weight out of range: %" "I64d", (long long)weights_e[k]);
```
Comment (DanGiles) [inline src/partition.c:1556]: Does the I64d still cause a problem in debug mode?

**Review by DanGiles — COMMENTED** [review]
Nice work @themkots! Just the one last comment on `I64d`. Happy to merge with our master then.

File: src/partition.c (hunk 1 above)
Comment (themkots) [reply src/partition.c:1556]: No. All 4 debug-enabled binaries [mpm,s32,s64,slocal] build OK with those format specifiers and the casting in those lines as per commit https://github.com/UCL/swiftsim/pull/27/commits/bb4d93b2da297d9bd82faea79f166ef0fe891574

**Pull request merged**